diff --git a/CMakeLists.txt b/CMakeLists.txt
index eec0993..e735f60 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,10 +43,16 @@ else()
       set(BLOCKDEV_TYPE  linux)
     endif()
     set (INSTALL_LIB 1)
-    add_definitions(-DCONFIG_HAVE_OWN_OFLAGS=0)
-    add_definitions(-DCONFIG_HAVE_OWN_ERRNO=0)
-    add_definitions(-DCONFIG_HAVE_OWN_ASSERT=0)
+
+    add_definitions(-DCONFIG_DEBUG_PRINTF=1)
+    add_definitions(-DCONFIG_DEBUG_ASSERT=1)
+
+    add_definitions(-DCONFIG_HAVE_OWN_OFLAGS=1)
+    add_definitions(-DCONFIG_HAVE_OWN_ERRNO=1)
+    add_definitions(-DCONFIG_HAVE_OWN_ASSERT=1)
+    add_definitions(-DCONFIG_USE_USER_MALLOC=0)
     add_definitions(-DCONFIG_BLOCK_DEV_CACHE_SIZE=16)
+
     add_subdirectory(fs_test)
 endif()
 
diff --git a/Makefile b/Makefile
index 12ee5c7..b68e850 100644
--- a/Makefile
+++ b/Makefile
@@ -19,6 +19,8 @@ COMMON_DEFINITIONS =                                      \
 	-DVERSION_MINOR=$(VERSION_MINOR)                      \
 	-DVERSION_PATCH=$(VERSION_PATCH)                      \
 	-DVERSION=$(VERSION)                                  \
+	-DLWEXT4_BUILD_SHARED_LIB=OFF \
+	-DCMAKE_INSTALL_PREFIX=./install \
 
 define generate_common
 	rm -R -f build_$(1)
@@ -29,6 +31,13 @@ define generate_common
 	-DCMAKE_TOOLCHAIN_FILE=../toolchain/$(1).cmake ..
 endef
 
+ARCH ?= x86_64
+#Output: src/liblwext4.a
+musl-generic:
+	$(call generate_common,$@)
+	cd build_$@ && make lwext4
+	cp build_$@/src/liblwext4.a ./liblwext4-$(ARCH).a
+
 generic:
 	$(call generate_common,$@)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 257b8cc..87665fc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,10 +4,12 @@ option(LWEXT4_BUILD_SHARED_LIB "Build shared library" OFF)
 #LIBRARY
 include_directories(.)
 aux_source_directory(. LWEXT4_SRC)
+set(M_SRC "../../ulibc.c")
+
 if(LWEXT4_BUILD_SHARED_LIB)
   add_library(lwext4 SHARED ${LWEXT4_SRC})
 else()
-  add_library(lwext4 STATIC ${LWEXT4_SRC})
+  add_library(lwext4 STATIC ${LWEXT4_SRC} ${M_SRC})
 endif()
 
 if  (DEFINED SIZE)
